### ISSUE TITLE
Match original Smart Moving feature-set

### ISSUE THREAD

**Issue by Tommsy64** [Tommsy64/SmartMovingReloaded#1]
### Movement
- [ ] Sprinting (vs. vanilla running)
- [ ] Strafe jumping
- [ ] Charged jump
- [ ] Wall jump
- [ ] Climb jump (What is this exactly...? Jumping up and back while climbing?)
- [ ] Headfirst jumping
- [ ] Crawling
- [ ] Sliding
- [ ] Alternate flying
- [ ] Alternate swimming
- [ ] Alternate ladder climbing
    - Differing speeds depending on ladder coverage
- [ ] Free Climbing
- [ ] Ceiling climbing
- [ ] Vine climbing

#### Other

- [ ] Toggelable sneaking
- [ ] Configurable everything

**Comment by PhysicalNova** [comment]
I must thank you. This is a courageous act to rewrite a abandonned mod. I wish you a lot of luck. Is the developpement going well ?

**Comment by SalmonSays** [comment]
This is massively exciting! Is there any chance you'd consider releasing an alpha with just crawling implemented since that was 80% of the appeal of the original Smart Moving for me and my users?

**Comment by XvBUSHvX** [comment]
Just wanted to take a moment to show my appreciation and a show of support. This has always helped minecraft to be a very immersive experience.

**Comment by Ariannae** [comment]
Echoing the other sentiments here. I absolutely loved what Smart Moving did for previous versions of Minecraft and would love to see it in a good state for 1.12.2 as well.

**Comment by Kraahkan** [comment]
you are awesome for attempting this

**Comment by rapus95** [comment]
i guess you can sort of cross alternate swimming/diving from this list since the new minecraft version supports it (i guess this makes some other stuff easier aswell like crawling)

**Comment by Ulisse54** [comment]
I seem to understand that no .jar file has yet been released to the public. Wish you luck, can't wait to be able to crawl in vents again.

**Comment by Tommsy64** [comment]
Thank you all for your support! There has been a hiatus, but I will be continuing development.

@Ulisse54 You can actually grab an alpha release over at the [releases page](https://github.com/Tommsy64/SmartMovingReloaded/releases). At the moment only crawling is implemented, but that's all you need to go exploring your vents again!

**Comment by SalmonSays** [comment]
@Tommsy64 For reference the V0.0.2 release has been broken which is why I forked it a few weeks ago. With a few very minor changes to build.gradle and mixins.smartmoving.core.json it actually loads and works on 1.12.2. I've been using this version to test:

https://github.com/SalmonSays/SmartMovingReloaded/releases

The big problem I've been trying to wrap my head around is the armor/player rendering pipeline. This problem has existed for a long while dating back to the original Smart Moving because of the way certain mods access net.minecraft.client.model.ModelBiped vs local methods I think? Armor from major mods like Botania is stuck facing North and doesn't match the body when crouching/crawling.

**Comment by Tommsy64** [comment]
@SalmonSays I've created an issue for the duplicate mod issue [here](https://github.com/Tommsy64/SmartMovingReloaded/issues/4).

**Comment by Ulisse54** [comment]
Oh perfect!

**Comment by beanaboston** [comment]
@rapus95 This mod is for 1.12.2, so alternate swimming should still be added, it would make sense to remove it for a hypothetical 1.13 version but should not be removed just cause Mojang added it in a new version.

The swimming is one of the biggest reasons I loved this mod, that and crawling. So if we don't get swimming later on in development I have a feeling a lot of people are going to be disappointed.

**Comment by Ulisse54** [comment]
@beanaboston I agree! But when this mod gets to 1.13.x a nice feature would be to let the player swim in places that are only 1 block deep, as it can't be done now.

**Comment by NinjaSamuraiX** [comment]
Welp, bookmarking this. Thanks for bringing back the mod! Excited to follow dev

**Comment by ParagonAries** [comment]
Hey! Good on you for attempting to revive an once dead mod! I used to use this mod all the time in the past. I hope it gets to 1.13.x soon!

**Comment by ScienceDrummer** [comment]
Getting kind of nervous as I haven't seen a recent commit. Last one was in September and we haven't really heard from Tommsy64 either.

**Comment by Tommsy64** [comment]
@ScienceDrummer Don't worry, I am still alive! I've been extremely busy with life for the past few months but still fully intend on completing and maintaining this mod. I've been active over on the Mo'Bends discord server, learning how to effectively mod the rendering of the player model in a way that does not break compatibility with other mods.

Keep in mind that working on a mod like this (at least for someone like me, with no real prior modding experience nor graphics programming/rendering experience) is extremely time consuming. I've spent a couple hundred hours getting Smart Moving Reloaded to its current state and will have to spend at least a few hundred more to implement the rest of the features.

**Comment by ScienceDrummer** [comment]
Oh, OK @Tommsy64, that's what I figured was going on. Good luck with the mod. :D

**Comment by ScienceDrummer** [comment]
F

**Comment by Arbifoxx** [comment]
Is this still happening?
